cpp/src/io/parquet/page_enc.cu

```diff
@@ -154,14 +158,19 @@ __global__ void __launch_bounds__(block_size) gpuInitPageFragments(PageFragment
     // etc...
     s->start_value_idx      = start_row;
     size_type end_value_idx = start_row + s->frag.num_rows;
-    for (size_type i = 0; i < s->col.nesting_levels; i++) {
-      s->start_value_idx = s->col.nesting_offsets[i][s->start_value_idx];
-      end_value_idx      = s->col.nesting_offsets[i][end_value_idx];
+    {
+      auto col = *(s->col.parent_column);
+      while (col.type().id() == type_id::LIST) {
+        auto offset_col = col.child(lists_column_view::offsets_column_index);
+        s->start_value_idx = offset_col.element<size_type>(s->start_value_idx);
+        end_value_idx = offset_col.element<size_type>(end_value_idx);
+        col = col.child(lists_column_view::child_column_index);
+      }
     }
     s->frag.start_value_idx = s->start_value_idx;
     s->frag.num_leaf_values = end_value_idx - s->start_value_idx;
 
-    if (s->col.nesting_levels > 0) {
+    if (s->col.parent_column != nullptr) {
```

Comment:
While this is correct because we don't generate `level_offsets` for non-nested. Could you change it to `if (s->col.level_offsets != nullptr)`. My bad.